db/fixtures/ae_datastore/ManageIQ/Cloud/Orchestration/Provisioning/StateMachines/Methods.class/__methods__/check_provisioned.rb

```diff
@@ -1,70 +1,113 @@
-  else
-    # deployment not done yet in provider
-    $evm.root['ae_result']         = 'retry'
-    $evm.root['ae_retry_interval'] = '1.minute'
-    return
-  end
+              private
 
-  $evm.log("info", "Stack deployment finished. Status: #{$evm.root['ae_result']}, reason: #{$evm.root['ae_reason']}")
-  $evm.log("info", "Please examine stack resources for more details") if $evm.root['ae_result'] == 'error'
+              def refresh_provider(service)
+                provider = service.orchestration_manager
 
-  return unless service.orchestration_stack
-  $evm.set_state_var('deploy_result', $evm.root['ae_result'])
-  $evm.set_state_var('deploy_reason', $evm.root['ae_reason'])
+                @handle.log("info", "Refreshing provider #{provider.name}")
+                @handle.set_state_var('provider_last_refresh', provider.last_refresh_date.to_i)
+                provider.refresh
+              end
 
-  refresh_provider(service)
+              def refresh_may_have_completed?(service)
+                stack = service.orchestration_stack
+                refreshed_stack = @handle.vmdb(:orchestration_stack).find_by(:name    => stack.name,
+                                                                             :ems_ref => stack.ems_ref)
+                refreshed_stack && refreshed_stack.status != 'CREATE_IN_PROGRESS'
+              end
 
-  $evm.root['ae_result']         = 'retry'
-  $evm.root['ae_retry_interval'] = '30.seconds'
-end
+              def check_deployed(service)
+                @handle.log("info", "Check orchestration deployed")
+
+                return unless deployment_completed?(service)
+                @handle.log("info", "Stack deployment finished. Status: " \
+                                    "#{@handle.root['ae_result']}, reason: #{@handle.root['ae_reason']}")
+
+                if @handle.root['ae_result'] == 'error'
+                  @handle.log("info", "Please examine stack resources for more details")
+                end
+
+                return unless service.orchestration_stack
+                state_and_root_vars(service)
+              end
+
+              def deployment_completed?(service)
+                # check whether the stack deployment completed
+                status, reason = service.orchestration_stack_status
+                case status.downcase
+                when 'create_complete'
+                  @handle.root['ae_result'] = 'ok'
+                when 'rollback_complete', 'delete_complete', /failed$/, /canceled$/
+                  @handle.root['ae_result'] = 'error'
+                  @handle.root['ae_reason'] = reason
+                else
+                  # deployment not done yet in provider
+                  @handle.root['ae_result']         = 'retry'
+                  @handle.root['ae_retry_interval'] = '1.minute'
+                  return false
+                end
+                true
+              end
+
+              def state_and_root_vars(service)
```

Comment:
This method name is not meaningful. The main logic in this method to prepare and call provider refresh.